javascript/ql/src/semmle/javascript/frameworks/Babel.qll

```diff
@@ -28,6 +28,18 @@ module Babel {
         result.(JSONArray).getElementStringValue(0) = pluginName
       )
     }
+
+    /**
+     * Gets a file affected by this Babel configuration.
+     */
+    Container getAContainerInScope() {
```

Comment:
We already have an `appliesTo` predicate in a subclass of `Config`; maybe we should just PUT that predicate here and improve it?

Reply:
Hadn't noticed that, though it doesn't appear in a subclass, it appears in the other plugin class.

I've introduced a `Plugin` class to factor out some common logic in the two plugin classes. We now also have `appliesTo` on both `Config` and `Plugin`.

The `getAContainerInScope()` has to be there in some form or other, since the recursion steps through folders, which aren't top-levels. We can rename it if you like. I'd like to keep it public, though, so we can easily tweak it at customers.